runtime/tests/decimal_operations.c

```diff
@@ -0,0 +1,49 @@
+#include <string.h>
+#include <assert.h>
+#include "../balrt.h"
+#include "../hash.h"
+#include "../third-party/decNumber/decQuad.h"
+
+HASH_DEFINE_KEY;
+
+void validate(TaggedPtr tp, const char *decStr) {
+    char str[DECQUAD_String];
+    decQuadToString((decQuad *)taggedToPtr(tp), str);
+    assert(strcmp(decStr, str) == 0);
+}
+
+void genAndValidateConstDec(const char *decStr, const char *dec) {
+    TaggedPtr tp = _bal_decimal_const(decStr);
+    validate(tp, dec);
+}
+
+void testConst() {
+    genAndValidateConstDec("1", "1");
+    genAndValidateConstDec("543", "543");
+    genAndValidateConstDec("1234567890123456789012345678901234", "1234567890123456789012345678901234");
+    genAndValidateConstDec("12345678901234567890123456789012345", "1.234567890123456789012345678901234E+34");
+    genAndValidateConstDec("123456789123456789123456789123456789", "1.234567891234567891234567891234568E+35");
+    genAndValidateConstDec("99999999999999999999999999999999999", "1.000000000000000000000000000000000E+35");
+    genAndValidateConstDec("123456789123456789123456789123456750", "1.234567891234567891234567891234568E+35");
+}
+
+void genAndValidateDecAdd(const char *decStr1, const char *decStr2, const char *sum) {
+    TaggedPtr tp1 = _bal_decimal_const(decStr1);
+    TaggedPtr tp2 = _bal_decimal_const(decStr2);
+    TaggedPtrPanicCode tp = _bal_decimal_add(tp1, tp2);
```

Comment:
Need to assert that the panicCode is zero.

Reply:
Changed